ReactSkia/modules/platform/libcurl/RSkNetworkingModule.cpp

```diff
@@ -73,125 +114,168 @@ jsi::Value RSkNetworkingModule::sendRequest(
+         curl_easy_setopt(curl, CURLOPT_CAINFO, CA_CERTIFICATE);
+         curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,0);
+    }
+    if(timeout == 0) {
+        /* if timeout is not specified, setting default time as 10L */
+        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 10L);
+    }
+    else
+        curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeout);
+    
+    curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);//Enable curl logs
 
-        if(method.compare("GET") == 0) {
-            /* send all data to this function  */
-            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteMemoryCallback);
+    // Progress callback and user data
+    curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0L); // Enable progress callback
+    curl_easy_setopt(curl, CURLOPT_PROGRESSDATA, networkRequest );
+    curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION, progressCallback);
+     
+    //Header callback and user data
+    curl_easy_setopt(curl, CURLOPT_WRITEHEADER, networkRequest);
+    curl_easy_setopt(curl, CURLOPT_HEADERFUNCTION, headerCallback);
 
-            /* we pass our 'chunk' struct to the callback function */
-            curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *)&chunk);
-        }
-        res = curl_easy_perform(curl);
-        /* Check for errors */
-        if(res != CURLE_OK) {
-            if(res == CURLE_OPERATION_TIMEDOUT) {
-                sendEventWithName("didCompleteNetworkResponse", folly::dynamic::array(requestId ,curl_easy_strerror(res), true));
-            }else {
-                sendEventWithName("didCompleteNetworkResponse", folly::dynamic::array(requestId ,curl_easy_strerror(res), false));
-            }
-            RNS_LOG_ERROR (stderr << "curl_easy_perform() failed: %s\n" <<curl_easy_strerror(res));
-        }else {
-            sendEventWithName("didReceiveNetworkData", folly::dynamic::array(requestId ,chunk.memory));
+    if(method.compare("POST") == 0) {
+      if(preparePostRequest(networkRequest, headers, data) == false)
+        return jsi::Value((int)CURL_RETURN_FAILURE);
+    } else if(method.compare("GET") == 0) {
+        // ResponseWrite callback and user data
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, networkRequest);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeMemoryCallback);
+    } else {
+        RNS_LOG_ERROR ("Unknown method\n") ;
+        return jsi::Value((int)CURL_RETURN_FAILURE);
+    }
+
+    // Add the request objet to list and notify app using callback function. 
+    connectionList_[networkRequest->requestId_] = networkRequest;
+    if(callbackObj.isFunction(rt)) {
+      jsi::Function callback = callbackObj.getFunction(rt);
+      callback.call(rt, (int) networkRequest->requestId_, 1);
+    }
+    res = curl_easy_perform(curl);
+    {
+      // completionBlock
+      if(networkRequest->curl_ != NULL) {
+        std::scoped_lock lock(isMutating);
+        if(!(networkRequest->useIncrementalUpdates_ && (networkRequest->responseType_ == "text")))
```

Comment:
is the responseType check required here ? since in sendData we are already checking it